### DISCUSSION TITLE
can't use mixins in theme components ?

### DISCUSSION THREAD

**Discussion by mlarcher** [mantinedev/mantine#5103] [Q&A]
I am trying to give default styling values to some components through the theme definition.
I seems it works when using hardcoded values, but not when I use the  `light-dark` mixin.
Is it intentional ? Is there a workaround ?
Here is the relevant code in my app :
```jsx
import 'normalize.css/normalize.css';
import '@mantine/core/styles.css';
import './globalStyles.css';

import { ColorSchemeScript, createTheme, MantineProvider, Popover } from '@mantine/core';
import Head from 'next/head';

const selectEffects = {
  border: `solid 1px light-dark(var(--mantine-color-dark-6), var(--mantine-color-dark-3))`,
  boxShadow: `0px 5px 12px 3px light-dark(var(--mantine-color-gray-5), var(--mantine-color-gray-7))`,
};

const theme = createTheme({
  components: {
    Popover: Popover.extend({
      styles: () => ({
        dropdown: selectEffects,
      }),
    }),
  },
});

function App({ Component, pageProps }) {
  return (
    <>
      <Head>
        <meta name="viewport" content="minimum-scale=1, width=device-width, initial-scale=1, shrink-to-fit=no" />
        <title>TPC</title>
      </Head>
      <ColorSchemeScript />
      <MantineProvider theme={theme}>
        <Component {...pageProps} />
      </MantineProvider>
    </>
  );
}

export default App;
```

**Comment by rtivital** [comment]
All postcss features including mixins and functions are only applied to the code in `*.css` files, they do not work in js/ts files.

**Reply by mlarcher** [reply]
damn, I was under the impression that they would work at least in style/styles too. Thanks for the heads up !

**Reply by mlarcher** [reply]
Just so you know, the migration from v6 to v7 is really tough. 
Lots of things are not really documented in the migration guide, and the styling changes have a lot of corner cases. 

And some other changes tricked me up too, like for example the multiselect not allowing react nodes as labels anymore, grouped options more complex in Selects, unexpected changes in AppShell, some attributes removed and no insight on how to get equivalent behaviour, etc.

I assume it's for the best, but I would have appreciated a little more guidance on the matter 🤷‍♂️
